reference-implementation/lib/writable-stream.js

```diff
@@ -505,16 +505,6 @@ class WritableStreamDefaultWriter {
       return Promise.reject(defaultWriterBrandCheckException('write'));
     }
 
-    const stream = this._ownerWritableStream;
```

Comment:
I moved the stream check back into write().

We need to check `state` after strategy.size() returns, since it may have changed. We could additionally check it before, if we wanted to guarantee that size() is only called in the writable state, but there are no tests for that and I didn't consider it important.

I have started thinking of size() as part of the public API. This makes the line between public methods and abstract operations a bit fuzzier.